clients/amqp-common/src/main/java/org/eclipse/hono/client/amqp/RequestResponseClient.java

```diff
@@ -363,7 +365,10 @@ private void handleResponse(final ProtonDelivery delivery, final Message message
                     span.log("response from peer accepted");
                 }
                 handler.one().handle(Future.succeededFuture(response));
-                ProtonHelper.accepted(delivery, true);
+                if (!delivery.isSettled()) {
+                    LOG.debug("client provided response handler did not settle message, auto-accepting ...");
+                    ProtonHelper.accepted(delivery, true);
```

Comment:
With the new northbound client, the delivery of telemetry and event messages can be settled by  the response handler. As command response is also a downstream message in par with telemetry and event, I have used the same approach in case of command response too. Currently we are using this feature any where.